src/extensions/persist.ts

```diff
@@ -0,0 +1,56 @@
+import debounce from 'lodash/debounce'
+import { DATABASE_INSTANCE, FactoryAPI } from '../glossary'
+import { isBrowser, supports } from '../utils/env'
+
+type ExtensionOption = {
+  storage?: Pick<Storage, 'getItem' | 'setItem'>
+  keyPrefix?: string
+}
+
+const STORAGE_KEY_PREFIX = 'mswjs-data'
+
+// Timout to persist state with some delay
+const DEBOUNCE_PERSIST_TIME_MS = 10
+
+/**
+ * Persist database in session storage
+ */
+export function persist(
+  factory: FactoryAPI<any>,
+  options: ExtensionOption = {},
+) {
+  if (!isBrowser() || (!options.storage && !supports.sessionStorage())) {
```

Comment:
Would be great if server-side persistence could be supported too.  I have that need in my project.  I want to serialise my mock DB and store it in Redis

Reply:
If @kettanaito will ok with this API, I could change interface for custom storage